### ISSUE TITLE
Ubuntu 12.04 installation

### ISSUE THREAD

**Issue by ReifiedException** [chmouel/ftp-cloudfs#45]
Trying to run ftp-cloudfs on Ubuntu 12.04 I have the following error messages:

/usr/local/lib/python2.7/dist-packages/pyftpdlib-1.0.1-py2.7.egg/pyftpdlib/ftpserver.py:54: DeprecationWarning: pyftpdlib.ftpserver module is deprecated
_depwarn("pyftpdlib.ftpserver module is deprecated")
/usr/local/lib/python2.7/dist-packages/pyftpdlib-1.0.1-py2.7.egg/pyftpdlib/ftpserver.py:59: DeprecationWarning: CallLater is deprecated; use pyftpdlib.ioloop.IOLoop.instance().call_later() instead
_depwarn("CallLater is deprecated; use "

As far as I can see, there is not any 'standard' python-pyftpdlib package for Ubuntu 12.04. This one was installed during setup.py run.

**Comment by reidrac** [comment]
I think this is because pyftpdlib 1.0.0 (released 2013-02-19) changed the code layout adding new stuff and deprecating pyftpdlib.ftpserver.

See: http://code.google.com/p/pyftpdlib/source/browse/trunk/HISTORY#97

It is not a big deal by now but I'll update the server to use the new pyftpdlib and bump the requirement to 1.0.0 or later.

Thanks for the report!

**Comment by reidrac** [comment]
Btw, Ubuntu 12.04 comes with python-pyftpdlib 0.6.0. You can install it with apt-get before running setup and use the pyftpdlib packaged by the distribution.

**Comment by ReifiedException** [comment]
Nope, as far as I can see it was first included to 12.10.

On Thu, Mar 7, 2013 at 12:10 PM, Juan J. Martínez
notifications@github.comwrote:

> Btw, Ubuntu 12.04 comes with python-pyftpdlib 0.6.0. You can install it
> with apt-get before running setup and use the pyftpdlib packaged by the
> distribution.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/chmouel/ftp-cloudfs/issues/45#issuecomment-14551803
> .

## 

## 

Those are my principles, and if you don't like them... well, I have others.
Groucho Marx

**Comment by reidrac** [comment]
Not that is important, but I'm using 12.04 and python-pyftpdlib was available. I believe it was introduced in 10.10.

```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 12.04.2 LTS
Release:    12.04
Codename:   precise
```

```
$ apt-cache show python-pyftpdlib
Package: python-pyftpdlib
Source: pyftpdlib
Version: 0.6.0-1
Architecture: all
Maintainer: Giampaolo Rodola' <g.rodola@...>
Installed-Size: 300
Depends: python2.7 | python2.6, python (>= 2.7.1-0ubuntu2)
Breaks: python (>= 2.8), python (<< 2.6)
Priority: optional
Section: python
Filename: pool/ubuntu/p/pyftpdlib/python-pyftpdlib_0.6.0-1_all.deb
Size: 47264
SHA256: cc72dc78485b5ad04afb5cf66c897729241856b0939cf0369f1cd8a747a46c16
SHA1: 45035575d0c718f59ef582bd38540197f069de4a
MD5sum: 4b6b2ae048d7fc3df277b1ffd82490f1
Description: High-level asynchronous FTP server library
 Python FTP server library provides an high-level portable interface to easily write asynchronous FTP servers with Python.
```

**Comment by ReifiedException** [comment]
Not that is so important but still...

ftpcloudfs% sudo lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 12.04.2 LTS
Release:        12.04
Codename:       precise
ftpcloudfs% apt-cache show python-pyftpdlib
N: Can't select versions from package 'python-pyftpdlib' as it is purely
virtual
N: No packages found
ftpcloudfs% apt-cache search python-pyftpdlib
ftpcloudfs%

(Fresh installed and updated Ubuntu instance in Essex).

On Thu, Mar 7, 2013 at 4:34 PM, Juan J. Martínez
notifications@github.comwrote:

> Not that is important, but I'm using 12.04 and python-pyftpdlib was
> available. I believe it was introduced in 10.10.
> 
> $ lsb_release -a
> No LSB modules are available.
> Distributor ID: Ubuntu
> Description:    Ubuntu 12.04.2 LTS
> Release:    12.04
> Codename:   precise
> 
> $ apt-cache show python-pyftpdlib
> Package: python-pyftpdlib
> Source: pyftpdlib
> Version: 0.6.0-1
> Architecture: all
> Maintainer: Giampaolo Rodola' g.rodola@...
> Installed-Size: 300
> Depends: python2.7 | python2.6, python (>= 2.7.1-0ubuntu2)
> Breaks: python (>= 2.8), python (<< 2.6)
> Priority: optional
> Section: python
> Filename: pool/ubuntu/p/pyftpdlib/python-pyftpdlib_0.6.0-1_all.deb
> Size: 47264
> SHA256: cc72dc78485b5ad04afb5cf66c897729241856b0939cf0369f1cd8a747a46c16
> SHA1: 45035575d0c718f59ef582bd38540197f069de4a
> MD5sum: 4b6b2ae048d7fc3df277b1ffd82490f1
> Description: High-level asynchronous FTP server library
>  Python FTP server library provides an high-level portable interface to easily write asynchronous FTP servers with Python.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/chmouel/ftp-cloudfs/issues/45#issuecomment-14563427
> .

## 

## 

Those are my principles, and if you don't like them... well, I have others.
Groucho Marx

**Comment by reidrac** [comment]
Looks like I got it from one of our internal repositories, sorry about that! :)

**Comment by eslng** [comment]
There's a workaround to install from 0. major version with pip to get rid of warnings. Just pip install pyftpdlib==0.7.0

**Comment by reidrac** [comment]
I've pushed a branch that supports pyftpdlib 1.2.0: https://github.com/chmouel/ftp-cloudfs/tree/pyftpdlib-1.2

Testing will be appreciated! :)

**Comment by reidrac** [comment]
I've released version 0.21 that supports pyftplib 1.2.0 (and hopefully later versions).
